openstudiocore/src/model/AirflowNetworkDistributionNode.cpp

```diff
@@ -0,0 +1,178 @@
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER, THE UNITED STATES GOVERNMENT, OR ANY CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ *  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *  AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **********************************************************************************************************************/
+
+#include "AirflowNetworkDistributionNode.hpp"
+#include "AirflowNetworkDistributionNode_Impl.hpp"
+#include "AirLoopHVACZoneMixer.hpp"
+#include "AirLoopHVACZoneMixer_Impl.hpp"
+#include "AirLoopHVACZoneSplitter.hpp"
+#include "AirLoopHVACZoneSplitter_Impl.hpp"
+#include "AirLoopHVACOutdoorAirSystem.hpp"
+#include "AirLoopHVACOutdoorAirSystem_Impl.hpp"
+#include "Node.hpp"
+#include "Node_Impl.hpp"
+
+#include <utilities/idd/IddFactory.hxx>
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/OS_AirflowNetworkDistributionNode_FieldEnums.hxx>
+
+#include "../utilities/units/Unit.hpp"
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  AirflowNetworkDistributionNode_Impl::AirflowNetworkDistributionNode_Impl(const IdfObject& idfObject,
+                                                                           Model_Impl* model,
+                                                                           bool keepHandle)
+    : AirflowNetworkNode_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == AirflowNetworkDistributionNode::iddObjectType());
+  }
+
+  AirflowNetworkDistributionNode_Impl::AirflowNetworkDistributionNode_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                                                           Model_Impl* model,
+                                                                           bool keepHandle)
+    : AirflowNetworkNode_Impl(other,model,keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == AirflowNetworkDistributionNode::iddObjectType());
+  }
+
+  AirflowNetworkDistributionNode_Impl::AirflowNetworkDistributionNode_Impl(const AirflowNetworkDistributionNode_Impl& other,
+                                                                           Model_Impl* model,
+                                                                           bool keepHandle)
+    : AirflowNetworkNode_Impl(other,model,keepHandle)
+  {}
+
```

Comment:
Should the nodes and component be children of this linkage?  If you delete the linkage should they also be deleted?